core/local/analysis.js

```diff
@@ -199,21 +199,23 @@ function analyseEvents (events /*: LocalEvent[] */, pendingChanges /*: LocalChan
             const addChange /*: ?LocalDirAddition */ = localChange.maybePutFolder(getChangeByInode(e))
             if (addChange) {
               changeFound(localChange.dirMoveFromAddUnlink(addChange, e))
-            } else if (getInode(e)) {
+              break
+            }
+            if (getInode(e)) {
               changeFound(localChange.dirDeletion(e))
-            } else {
-              const addChangeSamePath /*: ?LocalDirAddition */ = localChange.maybePutFolder(getChangeByPath(e))
-              if (addChangeSamePath && addChangeSamePath.wip) {
-                log.debug({path: addChangeSamePath.path, ino: addChangeSamePath.ino},
-                  'Folder was added then deleted. Ignoring add.')
-                // $FlowFixMe
-                addChangeSamePath.type = 'Ignored'
-              }
-
-              const moveChangeSamePath /*: ?LocalDirMove */ = localChange.maybeMoveFolder(getChangeByPath(e))
-              if (moveChangeSamePath && moveChangeSamePath.wip) {
-                localChange.convertDirMoveToDeletion(moveChangeSamePath)
-              }
+              break
+            }
+            const addChangeSamePath /*: ?LocalDirAddition */ = localChange.maybePutFolder(getChangeByPath(e))
```

Comment:
Why not using `withChangeByPath` here?

Reply:
I believe we're doing it. At least that's what the global diff shows. Maybe I made a mistake in some intermediate commit... Where did you see it?

Reply:
Please note that for some unknown reason existing additions & moves are checked in reverse order in the `unlinkDir` (vs the `unlink` one), which can be confusing regarding what is included in the `withChangeByPath` callback.